framework/mesh/mesh.h

```diff
@@ -56,9 +56,6 @@ class VolumeMesherPredefinedUnpartitioned;
 enum MeshAttributes : int
 {
   NONE = 0,
-  DIMENSION_1 = (1 << 0),
-  DIMENSION_2 = (1 << 1),
-  DIMENSION_3 = (1 << 2),
   ORTHOGONAL = (1 << 3),
   EXTRUDED = (1 << 4),
   UNSTRUCTURED = (1 << 5)
```

Comment:
To be honest, this whole enum is a bit weird. I'm not super familiar with how these flags work, but couldn't someone set the mesh attributes to both orthogonal and unstructured here? We should rework this and maybe have an enum class for mesh type as `ORTHOGONAL`/`UNSTRUCTURED` and a boolean attribute in whatever class for extruded.

Reply:
those bitwise operations. argh. I am sure they are useful but in what context? aren't those here equivalent to 1,2,4,8,16,32? why is bitwise better?

Reply:
1. I agree that this enum is weird as hell. I will be looking at removing it in future. Right now, I just undid the dimension non-sense.
2. one big thing that I think was a problem is that this way you can easily have mesh that is 1D, 2D and 3D at the same time (3 lower bits sets)

Reply:
I agree. I honestly don't see any good reason for us to ever use something like this.

Reply:
Bitwise operations are not in of themselves something to avoid but they should be used carefully. They are good when you want to specify a set of flags, which is probably why they were used her in the first place (i.e. dimension and orthogonal/unstructured are separate attributes). We use them for `SourceFlags` to specify sets of sources to `SetSource`. These are the `APPLY_WGS_SCATTER` type flags you see. In deal.ii, they are used to specify which unit cell matrices to construct. Here... It makes little sense though.

Reply:
I agree they have their place. I just don't see a necessary use case for us. It's too easy for inexperienced programmers to get things wrong (case in point... where we could have a mesh that's 1D, 2D, 3D all at the same time). For a code that may have a fair number of student programmers with limited experience, I'm just not a fan.

Reply:
What would you do with the `SourceFlags` then? Use a lot of boolean function arguments for `SetSource` routines?

Reply:
I haven't looked at the `SetSource` routines, so I can't offer an opinion. It could be they are the use case that pushes us to a bitset. If that turned out to be the case, it might make sense to write a flags class that makes it easier for our more inexperienced users. I've seen code where scoped enums are are used on top of a bitset to make something that works pretty well and hides all of the bitset complications.

Reply:
This is the call signature:
```
active_set_source_function_(groupset,
                            q_moments_local_,
                            phi_old_local_,
                            densities_local_,
                            APPLY_FIXED_SOURCES | APPLY_AGS_FISSION_SOURCES |
                              APPLY_WGS_FISSION_SOURCES);
```
The way it is used is quite user-readable and makes it easy to mix and match sources, which is particularly useful for the k-eigenvalue solvers. It would be good to keep a call signature as readable as this.